OptimizelySDK/Odp/OdpManager.cs

```diff
@@ -0,0 +1,296 @@
+            {
+                return false;
+            }
+
+            _odpConfig = newConfig;
+
+            // flush old events using old odp publicKey (if exists) before updating odp key.
+            // NOTE: It should be rare but possible that odp public key is changed for the same datafile (sdkKey).
+            // - Try to send all old events with the previous public key.
+            // - If it fails to flush all the old events here (network error), remaining events will be discarded.
+            EventManager.Flush();
+            EventManager.UpdateSettings(_odpConfig);
+
+            SegmentManager.ResetCache();
+            SegmentManager.UpdateSettings(_odpConfig);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="options">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        public string[] FetchQualifiedSegments(string userId, List<OdpSegmentOption> options)
+        {
+            if (SegmentManagerOrConfigNotReady())
+            {
+                _logger.Log(LogLevel.ERROR, Constants.ODP_NOT_ENABLED_MESSAGE);
+                return null;
+            }
+
+            return SegmentManager.FetchQualifiedSegments(userId, options).ToArray();
+        }
+
+        /// <summary>
+        /// Send identification event to ODP for a given full-stack User ID
+        /// </summary>
+        /// <param name="userId">User ID to send</param>
+        public void IdentifyUser(string userId)
+        {
+            if (EventManagerOrConfigNotReady())
+            {
+                _logger.Log(LogLevel.DEBUG, "ODP identify event not dispatched (ODP disabled).");
+                return;
+            }
+
+            if (!EventManager.IsStarted)
+            {
+                _logger.Log(LogLevel.DEBUG,
+                    "ODP identify event not dispatched (ODP not integrated).");
+                return;
+            }
+
+            EventManager.IdentifyUser(userId);
+        }
+
+        /// <summary>
+        /// Add event to queue for sending to ODP
+        /// </summary>
+        /// <param name="type">Type of event (typically `fullstack` from server-side SDK events)</param>
+        /// <param name="action">Subcategory of the event type</param>
+        /// <param name="identifiers">Key-value map of user identifiers</param>
+        /// <param name="data">Event data in a key-value pair format</param>
+        public void SendEvent(string type, string action, Dictionary<string, string> identifiers,
+            Dictionary<string, object> data
+        )
+        {
+            if (EventManagerOrConfigNotReady())
+            {
+                _logger.Log(LogLevel.DEBUG, "ODP event not dispatched (ODP disabled).");
```

Comment:
```suggestion
                _logger.Log(LogLevel.ERROR, "ODP event not dispatched (ODP disabled).");
```
Implicit event sending (IdentifyUser) should be DEBUG level, but when clients try to send events when ODP is not ready, it should be ERROR.